plasma_framework/test/src/exits/payment/PaymentChallengeIFEOutputSpent.test.js

```diff
@@ -0,0 +1,280 @@
+    const PIGGYBACK_BOND = 31415926535;
+    const MIN_EXIT_PERIOD = 60 * 60 * 24 * 7; // 1 week
+    const DUMMY_INITIAL_IMMUNE_VAULTS_NUM = 0;
+    const INITIAL_IMMUNE_EXIT_GAME_NUM = 1;
+    const OUTPUT_TYPE_ONE = 1;
+    const IFE_TX_TYPE = 1;
+    const OTHER_TX_TYPE = 2;
+    const ETH = constants.ZERO_ADDRESS;
+    const AMOUNT = 1;
+    const MERKLE_TREE_HEIGHT = 3;
+    const DUMMY_WITHDRAW_DATA = {
+        outputId: web3.utils.sha3('dummy output id'),
+        exitTarget: constants.ZERO_ADDRESS,
+        token: constants.ZERO_ADDRESS,
+        amount: 0,
+    };
+    const BLOCK_NUM = 1000;
+
+    before('deploy and link with controller lib', async () => {
+        const startInFlightExit = await PaymentStartInFlightExit.new();
+        const piggybackInFlightExit = await PaymentPiggybackInFlightExit.new();
+        const challengeInFlightExitNotCanonical = await PaymentChallengeIFENotCanonical.new();
+        const challengeIFEOutput = await PaymentChallengeIFEOutputSpent.new();
+
+        await PaymentInFlightExitRouter.link('PaymentStartInFlightExit', startInFlightExit.address);
+        await PaymentInFlightExitRouter.link('PaymentPiggybackInFlightExit', piggybackInFlightExit.address);
+        await PaymentInFlightExitRouter.link('PaymentChallengeIFENotCanonical', challengeInFlightExitNotCanonical.address);
+        await PaymentInFlightExitRouter.link('PaymentChallengeIFEOutputSpent', challengeIFEOutput.address);
+    });
+
+    before('deploy helper contracts', async () => {
+        this.exitIdHelper = await ExitId.new();
+        this.stateTransitionVerifierAccept = await StateTransitionVerifierAccept.new();
+    });
+
+
+    describe('challengeInFlightExitOutputSpent', () => {
+        const buildValidChallengeOutputArgs = async () => {
+            const output1 = new PaymentTransactionOutput(AMOUNT, alice, ETH);
+            const output2 = new PaymentTransactionOutput(AMOUNT, alice, ETH);
+            const inFlightTx = new PaymentTransaction(IFE_TX_TYPE, [0], [output1, output2]);
+            const inFlightTxBytes = web3.utils.bytesToHex(inFlightTx.rlpEncoded());
+
+            const merkleTree = new MerkleTree([inFlightTxBytes], MERKLE_TREE_HEIGHT);
+            const inclusionProof = merkleTree.getInclusionProof(inFlightTxBytes);
+
+            const outputId = computeNormalOutputId(inFlightTxBytes, 0);
+            const challengingTxOutput = new PaymentTransactionOutput(AMOUNT, alice, ETH);
+            const challengingTx = new PaymentTransaction(IFE_TX_TYPE, [outputId], [challengingTxOutput]);
+            const challengingTxBytes = web3.utils.bytesToHex(challengingTx.rlpEncoded());
+
+            const exitId = await this.exitIdHelper.getInFlightExitId(inFlightTxBytes);
+
+            const filler = web3.utils.sha3('filler');
+            const inFlightExit = {
+                exitStartTimestamp: (await time.latest()).toNumber(),
+                exitMap: 2 ** 4, // output with index 0 is piggybacked
```

Comment:
Minor: I would move 4 to const like `MAX_INPUT_SIZE`

Reply:
done